proparse/src/main/java/org/prorefactor/treeparser01/TP01Support.java

```diff
@@ -1126,20 +1154,20 @@ public void structorBegin(AST blockAST) {
 
   /** End of constructor or destructor. */
   @Override
-  public void structorEnd(AST blockAST) {
+  public void structorEnd(JPNode blockAST) {
     scopeClose(blockAST);
     currentRoutine = rootRoutine;
   }
 
   /** Called at the end of a VIEW statement. */
   @Override
-  public void viewState(AST headAST) {
+  public void viewState(JPNode headAST) {
     // The VIEW statement grammar uses gwidget, so we have to do some
     // special searching for FRAME to initialize.
-    JPNode headNode = (JPNode) headAST;
-    for (JPNode frameNode : headNode.query(NodeTypes.FRAME)) {
+    JPNode headNode = headAST;
+    for (JPNode frameNode : headNode.query(ProParserTokenTypes.FRAME)) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "query"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)